src/script/lua_api/l_craft.cpp

```diff
@@ -295,10 +295,10 @@ int ModApiCraft::l_clear_craft(lua_State *L)
 	CraftOutput c_output(output, 0);
 	if (!output.empty()) {
 		if (craftdef->clearCraftRecipesByOutput(c_output, getServer(L)))
-			return 0;
+			return 1;
```

Comment:
You return the craft table passed to `clear_craft` here. Did you mean to return `true`?

In which case it's:

```c++
lua_pushboolean(L, true);
return 1;
```

Reply:
Yes i meant to return true.
I did try using `lua_pushboolean()` throughout but it didn't work for some reason, i'll try again.